addons/utm/models/utm_mixin.py

```diff
@@ -74,6 +82,17 @@ def _find_or_create_record(self, model_name, name):
 
         return record
 
+    def find_or_create_record(self, model_name, name):
+        """ Version of ``_find_or_create_record`` used in frontend notably in
+        website_links. For UTM models it calls _find_or_create_record. For other
+        models (as through inheritance custom models could be used) it simply
+        calls a create.
+
+        :return: id of newly created or found record """
+        if model_name in self._tracking_models():
+            return self._find_or_create_record(model_name, name).id
+        return self.env[model_name].create({self.env[model_name]._rec_name: name}).id
```

Comment:
Why do we return the "id" specifically?

It's a little unusual and if compared to before, the JS call expected the raw return from "create".

Maybe I missed something.

Reply:
JS calls call_kw when using the "orm", and for create call_kw explicitly change the return type from a recordset to an id. As it is now a classic method, we have to manually return an ID. And api.returns does not work as model itself is dynamic.